numba/stencils/stencil.py

```diff
@@ -34,7 +35,7 @@ def __call__(self, context, builder, sig, args):
 @register_jitable
 def raise_if_incompatible_array_sizes(a, *args):
     ashape = a.shape
-    for arg in args:
+    for arg in literal_unroll(args):
```

Comment:
Would you please add a comment and probably include this example from the PR before this line just saying why the literal_unroll is necessary here.

Reply:
Done.